syncer/syncer.go

```diff
@@ -506,19 +509,14 @@ func (s *Syncer) reset() {
 func (s *Syncer) resetDBs() error {
 	var err error
 
-	// toDBs share the same `*sql.DB` in underlying `*baseconn.BaseConn`, currently the `BaseConn.ResetConn`
-	// can only reset the `*sql.DB` and point to the new `*sql.DB`, it is hard to reset all the `*sql.DB` by
-	// calling `BaseConn.ResetConn` once. On the other side if we simply change the underlying value of a
-	// `*sql.DB` by `*conn.DB = *db`, there exists some data race and invalid memory address in db driver.
-	// So we use the close and recreate way here.
-	closeConns(s.tctx, s.toDBs...)
-	s.toDBs, err = createConns(s.cfg, s.cfg.To, s.cfg.WorkerCount, maxDMLConnectionTimeout)
-	if err != nil {
-		return terror.WithScope(err, terror.ScopeDownstream)
+	for i := 0; i < len(s.toDBConns); i++ {
+		err = s.toDBConns[i].resetConn(s.tctx)
+		if err != nil {
+			return terror.WithScope(err, terror.ScopeDownstream)
+		}
 	}
-	s.tctx.L().Info("createDBs", zap.String("toDBs baseConn", fmt.Sprintf("%p", s.toDBs[0].baseConn.DB)))
 
-	err = s.ddlDB.ResetConn(s.tctx)
+	err = s.ddlDBConn.resetConn(s.tctx)
```

Comment:
should we need to reset connections for the shard group and online DDL?

Reply:
add sgk and onlineDDL [reset](https://github.com/pingcap/dm/pull/266/files#diff-ea4abf20044394891136e70431919a4fR519-R524), now it looks better